procedures/unit-testing-assertion-checks.ipf

```diff
@@ -216,8 +216,15 @@ static Function AreWavesEqual(wv1, wv2, mode, tol, detailedMsg)
 	string dimLabelMsg = ""
 	variable result, err
 
-	if((!numpnts(wv1) || !numpnts(wv2)) && mode == WAVE_DATA)
-		result = !numpnts(wv1) && !numpnts(wv2)
+	if(mode == WAVE_DATA)
+		if(!numpnts(wv1) || !numpnts(wv2))
+			result = !numpnts(wv1) && !numpnts(wv2)
+		elseif(WaveType(wv1, 1) != WaveType(wv2, 1))
+			detailedMsg = "Cannot compare data if the types don't match"
```

Comment:
I think the wording could be a bit more precise here. How about:
"The base wave types are different, WAVE_DATA can not be compared."

"base" related to text, numerical etc.
"WAVE_DATA" includes the specific compare constant .
The sentence has a better causality flow.

Reply:
I have changed the `detailMsg` to your suggestion. If you want to have the current types of both waves inserted we have to wait for PR #288 (Enhance error reporting in CHECK_WAVE, 2022-10-25).